lib/models/organization.ts

```diff
@@ -0,0 +1,216 @@
+/*
+Copyright 2020 Balena
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	 http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import * as errors from 'balena-errors';
+import * as Promise from 'bluebird';
+
+import * as BalenaSdk from '../../typings/balena-sdk';
+import { InjectedDependenciesParam } from '../balena';
+import {
+	isId,
+	isNotFoundResponse,
+	mergePineOptions,
+	treatAsMissingOrganization,
+} from '../util';
+
+const getOrganizationModel = function (deps: InjectedDependenciesParam) {
+	const { pine } = deps;
+
+	const getId = (handleOrId: string | number) =>
+		get(handleOrId, { $select: 'id' }).then(({ id }) => id);
```

Comment:
Using async/await on new code is best, but I guess you're limited by the ES version?

Reply:
The concern is that the transpiled code size will greatly increase after the async-degenerator code gets added.
We should though re-evaluate the cost of this vs the convenience, but I would avoid doing this now.

Reply:
Yeah, agreed, it only makes sense to switch when the minimum supported Node version has full async/await support.